java/private/checkstyle_config.bzl

```diff
@@ -36,6 +39,15 @@ checkstyle_config = rule(
             doc = "Additional files to make available to Checkstyle such as any included XML files",
             allow_files = True,
         ),
+        "checkstyle_binary": attr.label(
+            doc = "Checkstyle binary to use.",
+            default = "@contrib_rules_jvm//java:checkstyle_cli",
+            executable = True,
```

Comment:
Maybe consider adding either a macro that takes a checkstyle jar and automatically defines the `java_binary` or add an example to the docs that shows how to define it. I don't know if end users would be expected to know the location of checkstyle's main class, and it seems to be pretty stable across versions.

Reply:
Why not both? :)

Reply:
Done